util/postgres_db.py

```
@@ -0,0 +1,370 @@
+        It assumes that the data structure matches the table schema.
+
+        Args:
+            data (list): A list of dictionaries, where each dictionary represents a row to be
+                         inserted. The keys of the dictionary should match the column names
+                         in the table.
+
+        Returns:
+            bool: True if the insertion was successful, False otherwise.
+        """
+        if not self.connection:
+            raise ConnectionError("Database connection is not established.")
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)